docs/AUTH.md

```diff
@@ -0,0 +1,122 @@
+    
+    3rd Party App -->>3rd Party App : Generate a unique secret
+    3rd Party App ->>+HTTP Relay: Subscribe
+    note over 3rd Party App ,HTTP Relay: channel Id = hash(client secret)
+    3rd Party App ->>Authenticator: Show QR code
+    note over 3rd Party App ,Authenticator: required Capabilities,<br/>relay url, and client secret
+    Authenticator-->>User: Display consent form
+    User -->>Authenticator: Confirm consent
+    Authenticator-->>Authenticator: Sign AuthToken & encrypt with client secret
+    Authenticator->>HTTP Relay: Send encrypted AuthToken
+    note over Authenticator ,HTTP Relay: channel Id = hash(client secret)
+    HTTP Relay->>3rd Party App : Forward Encrypted AuthToken
+    HTTP Relay->>-Authenticator: Ok
+    3rd Party App -->>3rd Party App : Decrypt AuthToken & Resolve user's homeserver
+    3rd Party App ->>+Homeserver: Send AuthToken
+    Homeserver-->>Homeserver: Verify AuthToken
+    Homeserver->>-3rd Party App : Return SessionId
+    3rd Party App ->>+Homeserver: Request resources
+    Homeserver-->>Homeserver: Check Session capabilities
+    Homeserver ->>-3rd Party App: Ok
+```
+
+1. `3rd Party App` generates a unique (32 bytes) `client_secret`.
+1. `3rd Party App` uses the `base64url(hash(client_secret))` as a `channel_id` and subscribe to that channel on the `HTTP Relay` it is using.
+1. `3rd Party App` formats a Pubky Auth url:
+```
+pubkyauth:///
+   ?relay=<HTTP Relay base (without channel_id)>
+   &caps=<required capabilities>
+   &secret=<base64url(client_secret)>
+```
+ for example 
+ ```
+pubkyauth:///
+  ?relay=https://demo.httprelay.io/link
+  &caps=/pub/pubky.app/:rw,/pub/example.com/nested:rw
+  &secret=mAa8kGmlrynGzQLteDVW6-WeUGnfvHTpEmbNerbWfPI
+ ```
+ and finally show that URL as a QR code to the user.
+1. The `Authenticator` app scans that QR code, parses the URL and shows a consent form for the user.
+1. The user decides whether or not to grant these capabilities to the `3rd Party App`.
+1. If the user approves, the `Authenticator` uses their Keypair to sign an [`AuthToken`](#authtoken-encoding), then encrypts that token with the `client_secret`. The `channel_id` is then calculated by hashing that secret and the encrypted token is sent to the callback url, which is the `relay` + `channel_id`.
+1. `HTTP Relay` forwards the encrypted `AuthToken` to the `3rd Party App` frontend and confirms the delivery with the `Authenticator`.
+1. `3rd Party App` decrypts the AuthToken using its `client_secret`, reads the `pubky` in it and sends it to their `Homeserver` to obtain a session.
+1. `Homeserver` verifies the session and stores the corresponding `capabilities`.
+1. `Homeserver` returns a session Id to the frontend to use in subsequent requests.
+1. `3rd Party App` uses the session Id to access some resource at the Homeserver.
+1. `Homeserver` checks the session capabilities to see if it is allowed to access that resource.
+1. `Homeserver` responds to the `3rd Party App` with the resource.
```

Comment:
Numbers went off. But it is OK. I am changing this doc in my PR and IIRC it is already fixed there as a part of merge conflict resolution.